sql/src/main/java/io/crate/external/S3ClientHelper.java

```diff
@@ -86,28 +110,40 @@ protected AmazonS3 initClient(@Nullable String accessKey, @Nullable String secre
     }
 
     public AmazonS3 client(URI uri) throws IOException {
-        String accessKey = null;
-        String secretKey = null;
         if (uri.getHost() == null) {
             throw new IllegalArgumentException(INVALID_URI_MSG);
         }
+        // UserInfo should not be part of the URI as the AWS credentials should reside in the keystore
         if (uri.getUserInfo() != null) {
-            String[] userInfoParts = uri.getUserInfo().split(":");
-            try {
-                accessKey = userInfoParts[0];
-                secretKey = userInfoParts[1];
-            } catch (ArrayIndexOutOfBoundsException e) {
-                // ignore
+            if (!uri.getUserInfo().isEmpty()) {
+                throw new IllegalArgumentException("S3 credentials cannot be part of the URI but instead " +
+                                                   "must be stored inside the Crate keystore");
             }
+        } else if (uri.toString().contains("@") && uri.toString().contains(":")) {
```

Comment:
should we throw the same exception as above in this case too? we don't care about the URI being well formatted at this stage but only want to fail if the user attempts to provide the credentials

Reply:
I agree, its a malformatted URI and so should the exception tell one